src/main/java/org/pentaho/di/trans/dataservice/optimization/OptimizationImpactInfo.java

```diff
@@ -76,7 +86,7 @@ public String getDescription() {
       .append( "Step:  " )
       .append( getStepName() )
       .append( "\n" );
-    if ( getErrorMsg().length() > 0 ) {
+    if ( getErrorMsg() == null || getErrorMsg().length() > 0 ) {
```

Comment:
I have't changed this code, because the old code, which uses  setErrorMsg( String errorMsg ), may cause an error